Sources/NIOPosix/HappyEyeballs.swift

```diff
@@ -315,6 +349,19 @@ internal class HappyEyeballsConnector {
         return resolutionPromise.futureResult
     }
 
+    /// Initiate a DNS resolution attempt using Happy Eyeballs 2.
+    ///
+    /// returns: An `EventLoopFuture` that fires with a connected `Channel`.
+    @inlinable
+    func resolveAndConnect() -> EventLoopFuture<Channel> where ChannelBuilderResult == Void {
+        // We dispatch ourselves onto the event loop, rather than do all the rest of our processing from outside it.
+        self.loop.execute {
+            self.timeoutTask = self.loop.scheduleTask(in: self.connectTimeout) { self.processInput(.connectTimeoutElapsed) }
+            self.processInput(.resolve)
+        }
+        return resolutionPromise.futureResult.map { $0.0 }
+    }
```

Comment:
This function is just the same as the one above with `.map { $0.0 }` at the end, can we call that one and just chain this on?